cpp/src/io/parquet/predicate_pushdown.cpp

```diff
@@ -444,55 +309,180 @@ std::optional<std::vector<std::vector<size_type>>> aggregate_reader_metadata::fi
     columns.push_back(std::move(min_col));
     columns.push_back(std::move(max_col));
   }
-  auto stats_table = cudf::table(std::move(columns));
 
-  // Converts AST to StatsAST with reference to min, max columns in above `stats_table`.
-  stats_expression_converter const stats_expr{filter.get(),
-                                              static_cast<size_type>(output_dtypes.size())};
-  auto stats_ast     = stats_expr.get_stats_expr();
-  auto predicate_col = cudf::detail::compute_column(stats_table, stats_ast.get(), stream, mr);
-  auto predicate     = predicate_col->view();
-  CUDF_EXPECTS(predicate.type().id() == cudf::type_id::BOOL8,
-               "Filter expression must return a boolean column");
+  // Apply bloom filter membership to row groups for each equality predicate column, literal pair
+  // and get corresponding boolean columns.
+  auto [bloom_filter_membership_cols, equality_literals] =
```

Comment:
@bdice refactored to use the same table for both statsAST and bloomfilterAST evaluations altogether.

Reply:
CC: @vuule @karthikeyann

Reply:
Reverted the table fusion as it is possible that we don't have stats available (there is not really a way to tell), leading to an all false LOGICAL_AND between statsAST and bloomfilterAST.